### PR TITLE
Adds support for testcontainers

### PR DESCRIPTION
**Pull request by ashishkujoy**

### CODE REVIEW

File: kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/StartableTestLifecycleAware.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.kotest.extensions.testcontainers
+
+
+import org.testcontainers.lifecycle.TestDescription
+import org.testcontainers.lifecycle.TestLifecycleAware
+import java.util.*
+
+internal class StartableTestLifecycleAware : TestStartable(), TestLifecycleAware {
```
Comment (sksamuel) [inline kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/StartableTestLifecycleAware.kt:8]: What is the purpose of this class

File: kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package io.kotest.extensions.testcontainers
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class TestcontainerKotestIntegrationPerSpec : StringSpec({
+   val testStartable = TestStartable()
+   listeners(testStartable.perSpec())
```
Comment (sksamuel) [inline kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt:8]: In master I have changed listener to return the type passed in.
So we could do

```
val startable = listener(TestStartable().perSpec())
```

Another option is 
```
val startable = TestStartable().apply { listener(this) }
```

Comment (sksamuel) [reply kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt:8]: We could add a function to `TestConfiguration` like this:
```
   fun <T : TestListener> T.register(): T {
      listener(this)
      return this
   }
```

Then you could just do 

```
val startable = TestStartable().register()
```

But it does mean that you would have to have some way of deciding if the test startable is a per spec or per test. Perhaps

```
val startable = TestStartable().perSpec.register()
```

File: kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/StartableTestLifecycleAware.kt (hunk 1 above)
Comment (ashishkujoy) [reply kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/StartableTestLifecycleAware.kt:8]: This class is used for testing in two places. One in ```TestLifecycleAwareListenerTest``` and second in ```StartableTestLifecycleAwareTest```.
This class extends ```TestLifecycleAware``` in its before test i am storing the ```TestDescription``` passed to it. Later in  ```TestLifecycleAwareListenerTest```  i am checking that test description passed to it has fileSystemFriendly name and in ```StartableTestLifecycleAwareTest``` i am checking that if a startable is also a ```TestLifecycleAware``` then we are calling its beforeTest and afterTest in both perSpec and perTest variants.

File: kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt (hunk 2 above)
Comment (ashishkujoy) [reply kotest-extensions/kotest-extensions-testcontainers/src/jvmTest/kotlin/io/kotest/extensions/testcontainers/TestcontainerKotestIntegrationPerSpec.kt:8]: ```val startable = TestStartable().perSpec.register()```
Looks good to me.

**Comment by sksamuel** [comment]
If you want to make those changes you can.
Merge when you're happy, it looks good.

**Pull request merged**